### PR TITLE
Adds return type to the UI store's view

### PR DESCRIPTION
**Pull request by pratap0007**
This patch adds return type in the UI store's
view and does minor refactoring

Signed-off-by: Shiv Verma <shverma@redhat.com>

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
- [ ] Run API Unit Tests, Lint Checks, API Design, Golden Files with `make api-check`
- [x] Run UI Unit Tests, Lint Checks with `make ui-check`
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/main/CONTRIBUTING.md) for more details._

### CODE REVIEW

File: ui/src/containers/BasicDetails/index.tsx [hunk 1]
```diff
@@ -35,11 +35,10 @@ import TooltipDisplay from '../../components/TooltipDisplay';
 import Rating from '../Rating';
 import { titleCase } from '../../common/titlecase';
 import { assert } from '../../store/utils';
-import { ICatalogStore } from '../../store/catalog';
 import './BasicDetails.css';
 
 const BasicDetails: React.FC = () => {
-  const { resources, catalogs }: { resources: IResourceStore; catalogs: ICatalogStore } = useMst();
+  const { resources, catalogs } = useMst();
```
Comment (PuneetPunamiya) [inline ui/src/containers/BasicDetails/index.tsx:41]: Any reason to remove the types here

Comment (pratap0007) [reply ui/src/containers/BasicDetails/index.tsx:41]: we don't need it here , as I have mentioned type in the root store

File: ui/src/containers/Details/index.tsx [hunk 2]
```diff
@@ -25,7 +30,9 @@ const Details: React.FC = observer(() => {
   const resourceKey = `${catalog}/${titleCase(kind)}/${name}`;
 
   const validateUrl = () => {
-    const allVersions = resources.resources.get(resourceKey).versions;
+    const resource = resources.resources.get(resourceKey);
+    assert(resource);
```
Comment (PuneetPunamiya) [inline ui/src/containers/Details/index.tsx:34]: Not urgent but i think we need to get rid of this assert aka undefined values

**Comment by PuneetPunamiya** [comment]
Awesome 
Thanks for updating the types
/approve
/meow

**Pull request merged**